File: gateway-console/src/types/modelSpec.ts
/** 模型规格状态枚举 */
export type ModelSpecState = 'ACTIVE' | 'INACTIVE';

/** 模型规格信息（与后端 ModelSpecResponse 一致） */
export interface ModelSpec {
  id: number;
  /** 所属供应商 ID */
  providerId: number;
  /** 供应商模型标识（如 gpt-4o、claude-3-opus） */
  providerModelId: string;
  /** 显示名称 */
  displayName?: string;
  /** 模型族（如 gpt-4、claude-3） */
  modelFamily?: string;
  /** 上下文窗口大小 */
  contextWindow?: number;
  /** 最大输入 Token 数 */
  maxInputTokens?: number;
  /** 最大输出 Token 数 */
  maxOutputTokens?: number;
  /** 模型能力（如 vision、function_calling） */
  capabilities?: Record<string, boolean>;
  /** 支持的模态（如 text、image、audio） */
  modalities?: string[];
  /** 状态 */
  state: ModelSpecState;
  /** 优先级 */
  priority?: number;
  /** 权重 */
  weight?: number;
  createdAt: string;
  updatedAt: string;
}

/** 创建模型规格请求 */
export interface CreateModelSpecRequest {
  /** 所属供应商 ID */
  providerId: number;
  /** 供应商模型标识 */
  providerModelId: string;
  /** 显示名称 */
  displayName?: string;
  /** 模型族 */
  modelFamily?: string;
  /** 上下文窗口大小 */
  contextWindow?: number;
  /** 最大输入 Token 数 */
  maxInputTokens?: number;
  /** 最大输出 Token 数 */
  maxOutputTokens?: number;
  /** 模型能力 */
  capabilities?: Record<string, boolean>;
  /** 支持的模态 */
  modalities?: string[];
  /** 优先级 */
  priority?: number;
  /** 权重 */
  weight?: number;
}

/** 更新模型规格请求 */
export interface UpdateModelSpecRequest {
  /** 供应商模型标识 */
  providerModelId?: string;
  /** 显示名称 */
  displayName?: string;
  /** 模型族 */
  modelFamily?: string;
  /** 上下文窗口大小 */
  contextWindow?: number;
  /** 最大输入 Token 数 */
  maxInputTokens?: number;
  /** 最大输出 Token 数 */
  maxOutputTokens?: number;
  /** 模型能力 */
  capabilities?: Record<string, boolean>;
  /** 支持的模态 */
  modalities?: string[];
  /** 优先级 */
  priority?: number;
  /** 权重 */
  weight?: number;
  /** 状态 */
  state?: ModelSpecState;
}

File: gateway-boot/src/main/java/com/codingas/gateway/domain/supply/entity/ModelSpec.java
package com.codingas.gateway.domain.supply.entity;

import com.codingas.gateway.common.entity.BaseEntity;
import com.codingas.gateway.common.entity.DomainEntity;
import com.codingas.gateway.domain.supply.enums.ModelSpecState;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.extern.slf4j.Slf4j;

import java.util.List;
import java.util.Map;

/**
 * 模型规格实体（从 Model 拆出规格部分）
 *
 * <p>ModelSpec 是模型固有规格，与渠道无关。</p>
 */
@Data
@EqualsAndHashCode(callSuper = true)
@DomainEntity
@Slf4j
public class ModelSpec extends BaseEntity {

    
    /** 供应商侧标识（如 "gpt-4o"，路由匹配用） */
    private String providerModelId;

    private String displayName;

    private String modelFamily;

    private Integer contextWindow;

    private Integer maxInputTokens;

    private Integer maxOutputTokens;

    private Map<String, Boolean> capabilities;

    private List<String> modalities;

    private ModelSpecState state = ModelSpecState.ACTIVE;

    /** 路由优先级（数值越小优先级越高） */
    private Integer priority;

    /** 负载均衡权重 */
    private Integer weight;

    /**
     * 检查模型规格是否可用
     */
    public boolean isAvailable() {
        return ModelSpecState.ACTIVE.equals(state);
    }
}